src/praisonai/praisonai/scheduler/shared.py

```
@@ -25,25 +33,21 @@ def parse(schedule_expr: str) -> int:
-        if expr.startswith("cron:"):
+        # Wrapper-specific: cron → best-effort integer interval for polling loop.
+        if expr.lower().startswith("cron:"):
             return ScheduleParser._parse_cron_to_interval(expr[5:].strip())
+
+        # Delegate the shared grammar to the single core owner.
+        from praisonaiagents.scheduler.parser import parse_schedule
+
+        try:
+            sched = parse_schedule(expr)
+        except ValueError:
+            raise ValueError(f"Unsupported schedule format: {schedule_expr}")
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)